### ISSUE TITLE
Request: tools for drawing shapes

### ISSUE THREAD

**Issue by introlinux** [OpenBoard-org/OpenBoard#128]
Drawing shapes and arrows are really useful for organize ideas in documents. Now I import shape images, but it is an ugly way and the result is poor.

**Comment by celfred** [comment]
I know some of my colleagues would need such a feature. This prevents them from switching from OpenSabkoré to OpenBoard for the moment.

**Comment by ramuntcho** [comment]
OpenBoard is a simpler version than Open-Sankore, focused to the work of a teacher in a classroom, privileging the ease of use. 
In this case, this function doesn't seem essential and won't be added for the moment.

**Comment by mannaia** [comment]
I would also stress that such a feature (drawing shapes) can be an extremely useful feature. Hope this will be considered in next upgrades.

**Comment by omartek** [comment]
+1 from me too. Old Drawing tool very useful.

**Comment by jkinred** [comment]
Would very much like to see this too, as it would allow focus on the expression of ideas without having to concentrate on drawing a shape and losing the flow, which white boarding is so good for.

I am using OpenBoard as a whiteboard in a remote team and it's very good for this purpose.

Quite often I need to draw basic shapes -- like squares and circles. Even though I draw with a tablet I find myself having to do so more slowly to get a good result like I would on a real whiteboard.

**Comment by diegocaja** [comment]
I think something similar to the functionality "stroke recognizer" of Xournal++ or witeboard would be nice.

Another possibility could be a stylus bar similar to awwapp or why not have a selection of arrows, circles or rectangles in the draw line option of the stylus as Xournal++ app.

I really would appreciate it, if this functionality was added to OpenBoard.

**Comment by jkinred** [comment]
Subsequent to my last comment, I found the Shapes functionality in the right hand drawer which has SVG shapes. Not as nice as an actual drawing tool, but suits most of my needs.

**Comment by erralb** [comment]
It would be great to have at least the possibility to change the lines to arrows.
Like in many diagrams softwares, you can pick what you want to display for both ending of a line, arrow or nothing

**Comment by FedeCuci** [comment]
+1 for me as well. Would love to have this feature for basic shapes such as rectangles and circles.

**Comment by thewickedkarma** [comment]
Since it doesn't have tools for drawing specific shapes, it isn't of much help for teaching STEM subjects.

**Comment by FedeCuci** [comment]
I would love to be skilled enough to help implement this....

**Comment by SeeDborg** [comment]
I'm upping this topic because I have many demands of teachers who would like to bring back the old "shapes" tool from Open Sankoré. Would it be possible to think about adding this in a future version ?
For a reminder, it looked like this : 

<img width="749" height="575" alt="Image" src="https://github.com/user-attachments/assets/a674653e-2813-4827-aa4d-70c86a56b4e9" />

**Comment by kaamui** [comment]
Hi,

I remember this feature quite well, it's my baby, and was quite proud of it.  ^^

But it was not preserved in OpenBoard to stick to the simplicity ruling OpenBoard, as stated by my former colleague above.

And as it was never asked by our teachers internally (also probably because it is not known), I can't prioritize on it and retrieve the code from Sankoré.

But if somebody wants to give it a try, and migrate the code (written with Qt4, but a lot is non-Qt related if I remember well, so a good part of it can probably be reused as-is), with an option to disable it in the config file, I could merge it.

Sankoré's code is here : https://github.com/Sankore/Sankore-3.1

There's quite some commits about it, but an example to help dig in : https://github.com/Sankore/Sankore-3.1/commit/00c93e44d45bef734390f8226185d2c1a35660a6

You'll have to search commits related with "EV-7" mentioned, a good portion of the code being in the domain and gui folders, e-g `UBEditable` and its children, `UBAbstractHandle` and all its children, UBDrawingPalette, etc

**Comment by SeeDborg** [comment]
Thanks for the quick answer !

You're right to be proud, it was a great feature, and I know many people disappointed by its disappearance...

I don't have the knowledge to retrieve the code, but if you or someone else can adapt the code and we can merge it, it would be great news 👍

**Comment by SeeDborg** [comment]
Also I have a hard time to understand why such a feature disappeared from OpenBoard. It's not adding any difficulty to OpenBoard.

**Comment by Blankovitch** [comment]
> I'm upping this topic because I have many demands of teachers who would like to bring back the old "shapes" tool from Open Sankoré. Would it be possible to think about adding this in a future version ? For a reminder, it looked like this :
> 
Yes this feature was so useful. It would be great to see it coming back !

**Comment by rafgarnier-source** [comment]
Good idea. I miss this feature.

**Comment by fen30** [comment]
> I'm upping this topic because I have many demands of teachers who would like to bring back the old "shapes" tool from Open Sankoré. Would it be possible to think about adding this in a future version ? For a reminder, it looked like this :
> 
> <img alt="Image" width="749" height="575" src="https://private-user-images.githubusercontent.com/7523821/545528685-a674653e-2813-4827-aa4d-70c86a56b4e9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMDA1NTMsIm5iZiI6MTc3MDMwMDI1MywicGF0aCI6Ii83NTIzODIxLzU0NTUyODY4NS1hNjc0NjUzZS0yODEzLTQ4MjctYWE0ZC03MGM4NmE1NmI0ZTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDIwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAyMDVUMTQwNDEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2JhYmExZDg5ZmYyODdkZGUyMDA5NTU2Y2U1NWMwM2IyYjcyYzViYWI4YWExNTJmOWQzZTgyZDdiN2JhNTUzZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.dtv9y14d_b7ACgchv6QPWQEs4Ln8dP_kBp3emY_eTa4">

**Comment by celfred** [comment]
I wish I had the skills to help on that one… but I vote for it ! 👍

**Comment by letsfindaway** [comment]
> I don't have the knowledge to retrieve the code, but if you or someone else can adapt the code and we can merge it, it would be great news 👍

I think I could help here. But first, I want to stabilize the other features that have already been merged for 1.8.0: Wayland support, table of contents, and flexible backgrounds.

**Comment by SeeDborg** [comment]
Wow, thank you so much, it would be wonderful to see the option back in 1.8.0 !
